src/main/java/com/google/devtools/build/lib/analysis/BuildView.java

```diff
@@ -440,11 +459,11 @@ public AnalysisResult update(
     }
 
     Set<ConfiguredTarget> targetsToSkip =
-        new TopLevelConstraintSemantics(
-                skyframeExecutor.getPackageManager(),
-                input -> skyframeExecutor.getConfiguration(eventHandler, input),
-                eventHandler)
-            .checkTargetEnvironmentRestrictions(skyframeAnalysisResult.getConfiguredTargets());
+        ImmutableSet.copyOf(
+            Sets.union(
+                topLevelConstraintSemantics.checkTargetEnvironmentRestrictions(
+                    skyframeAnalysisResult.getConfiguredTargets()),
+                platformRestrictions.targetsToSkip));
```

Comment:
Nit: Replace "ImmutableSet.copyOf(Sets.union(...))" with the equivalent "Sets.union(...).immutableCopy()"

Reply:
Done